src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java

```diff
@@ -121,8 +121,8 @@ public AuthenticateResponse authenticate(final AuthenticationContext context, fi
                 Object msvAvFlags = challenge.getAvPairObject(AvId.MsvAvFlags);
                 if (msvAvFlags instanceof Long && ((long) msvAvFlags & 0x00000002) > 0) {
                     // MIC should be calculated
-                    NtlmAuthenticate resp = new NtlmAuthenticate(new byte[0], ntlmv2Response,
-                        context.getUsername(), context.getDomain(), null, sessionkey, EnumWithValue.EnumUtils.toLong(negotiateFlags),
+                    NtlmAuthenticate resp = new NtlmAuthenticate(new byte[24], ntlmv2Response,
```

Comment:
Why are we passing a 24 byte array in?

Reply:
ok will add a config for the workstation name. Regarding the byte[24], I don't recollect now, if this was really needed or not. But spend quite some time at customer env, where the authentication was failing. I can keep the byte[24] out of this PR

Just for FYI, We have been using this in production for a long time now, and has been verified to work great against NetApp, Dell/EMC, Isilon etc after some of the above fixes.